frontend/pages/admin/AppSettingsPage/AppSettingsPage.tsx

```diff
@@ -52,6 +52,8 @@ const AppSettingsPage = (): JSX.Element => {
   const onFormSubmit = useCallback(
     (formData: IConfigNested) => {
       const diff = deepDifference(formData, appConfig);
+      // send all formData.agent_options because diff overrides all agent options
+      diff.agent_options = formData.agent_options;
```

Comment:
Is the `"/api/v1/fleet/config` API supposed to work with diff updates?

@chiiph @mna

Reply:
I believe so, I have a bunch pf small yamls that I use to set specific things.

Reply:
Ah right, I now realize we load current (`appConfig, err := svc.AppConfig(ctx)`) and then apply the incoming fields only (`decoder.Decode(&appConfig)`).